src/crimson/osd/pg.cc

```diff
@@ -318,6 +318,7 @@ unsigned PG::get_target_pg_log_entries() const
 }
 
 void PG::on_removal(ceph::os::Transaction &t) {
+  clear_log_entry_maps();
```

Comment:
This shouldn't be necessary -- any PG invoking on_removal would have to have gone through on_change since the last time it was primary and placed anything in the map.  You can assert here if you like, but on_change should be the only place this needs to be cleared.